tsdownsample/_python/downsamplers.py

```diff
@@ -5,6 +5,29 @@
 from ..downsampling_interface import AbstractDownsampler
 
 
+def _get_bin_idxs(x: np.ndarray, nb_bins: int) -> np.ndarray:
+    """Get the equidistant indices of the bins to use for the aggregation.
+
+    Parameters
+    ----------
+    x : np.ndarray
+        The x values of the input data.
+    nb_bins : int
+        The number of bins.
+
+    Returns
+    -------
+    np.ndarray
+        The indices of the bins to use for the aggregation.
+    """
+    # Thanks to the `linspace` the data is evenly distributed over the index-range
+    # The searchsorted function returns the index positions
+    bins = np.searchsorted(x, np.linspace(x[0], x[-1], nb_bins + 1), side="right")
+    bins[0] = 0
+    bins[-1] = len(x)
+    return np.unique(bins)
```

Comment:
If you agree with this @jonasvdd, we can be pretty sure that the rust code is compliant with the Python code :handshake:

Reply:
I agree with this implementation! 🤝